skyportal/handlers/api/photometry.py

```diff
@@ -947,19 +947,26 @@ def delete(self, upload_id):
               application/json:
                 schema: Error
         """
-        # Permissions check:
-        phot_id = Photometry.query.filter(Photometry.upload_id == upload_id).first().id
-        _ = Photometry.get_if_readable_by(phot_id, self.current_user)
 
-        n_deleted = (
-            DBSession()
-            .query(Photometry)
-            .filter(Photometry.upload_id == upload_id)
-            .delete()
-        )
-        self.finalize_transaction()
+        # dont check permissions here -- pull all the photometry associated with
+        # the upload, not necessarily just the photometry that is accessible
+        # to the user. if any of the photometry fails to be deleted, send back
+        # 400
+        photometry_to_delete = Photometry.query.filter(
+            Photometry.upload_id == upload_id
+        ).all()
+
+        n = len(photometry_to_delete)
+        if n == 0:
+            return self.error('Invalid bulk upload id.')
 
-        return self.success(f"Deleted {n_deleted} photometry points.")
+        for phot in photometry_to_delete:
+            DBSession().delete(phot)
+
+        # this will raise if the user does not have access to delete any of
```

Comment:
We should catch raises and use self.error

Reply:
Thanks, yeah, that's what's done here - this raises an AccessError which triggers `self.error` in the exception handling hook and doesn't actually raise up the stack. I'll update the comment here to make that clearer